apps/cbportal/base/CBGrid/.macros/wiki/physical_node_data/1_physical_node_data.py

```diff
@@ -0,0 +1,45 @@
+
+def main(j, args, params, tags, tasklet):
+    params.result = (args.doc, args.doc)
+    nid = args.requestContext.params.get('id')
+    gid = args.requestContext.params.get('gid')
+    ccl = j.clients.osis.getNamespace('cloudbroker')
+    scl = j.clients.osis.getNamespace('system')
+    node = scl.node.searchOne({'id': int(nid) if nid else None})
+    data = {'node':None, 'stack': None, 'ovs': None}
+
+    # check node exists
+    if not node:
+        args.doc.applyTemplate(data, True)
+        return params
+    data['node'] = node
+
+    for netaddr in node['netaddr']:
+        if netaddr['name'] == 'storage':
+            data['storageip'] = netaddr['ip'][0]
+            break
+    else:
+        raise RuntimeError('Storage Node does not have interface for storage ip , please specify storage interface')
```

Comment:
no runtime errors in macros, instead make it print something on the page something usefull